### PR TITLE
Add MessageListAppearance data class

### PR DESCRIPTION
**Pull request by bhm**

### CODE REVIEW

File: app/ui/src/main/java/com/fsck/k9/ui/messagelist/MessageListAppearance.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.fsck.k9.ui.messagelist
+
+import com.fsck.k9.K9
+
+data class MessageListAppearance(
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/ui/messagelist/MessageListAppearance.kt]: I'd rather have this be a simple data class without a dependency to `K9`. Please pass the respective values when creating the instance instead.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 2]
```diff
@@ -284,41 +270,28 @@ class MessageListAdapter internal constructor(
...
                 Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
 
         // Set span (color) for preview message
-        previewText.setSpan(
-                ForegroundColorSpan(previewTextColor),
-                beforePreviewText.length + sigil.length,
-                previewText.length,
-                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
-        )
+        previewText.setSpan(ForegroundColorSpan(previewTextColor), beforePreviewText.length + sigil.length,
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Unrelated code style change.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 3]
```diff
@@ -284,41 +270,28 @@ class MessageListAdapter internal constructor(
...
         return AbsoluteSizeSpan(fontSize, true)
     }
 
-    private fun fetchCounterPartyAddress(
-            fromMe: Boolean,
-            toAddrs: Array<Address>,
-            ccAddrs: Array<Address>,
-            fromAddrs: Array<Address>
-    ): Address? {
+    private fun fetchCounterPartyAddress(fromMe: Boolean, toAddrs: Array<Address>, ccAddrs: Array<Address>, fromAddrs: Array<Address>): Address? {
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Unrelated code style change.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 4]
```diff
@@ -418,4 +391,4 @@ class MessageListAdapter internal constructor(
 interface MessageListItemActionListener {
     fun toggleMessageSelectWithAdapterPosition(position: Int)
     fun toggleMessageFlagWithAdapterPosition(position: Int)
-}
+}
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Unrelated code style change.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 5]
```diff
@@ -127,7 +113,6 @@ class MessageListAdapter internal constructor(
 
     var selected: Set<Long> = emptySet()
 
-
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: Unrelated code style change.

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt [hunk 6]
```diff
@@ -97,21 +98,6 @@ class MessageListAdapter internal constructor(
         array.recycle()
     }
 
-    private inline val previewLines: Int
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: :heart:

Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: ❤️

File: app/ui/src/main/java/com/fsck/k9/ui/messagelist/MessageListAppearance.kt (hunk 1 above)
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/ui/messagelist/MessageListAppearance.kt]: done

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt (hunk 3 above)
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: done

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt (hunk 4 above)
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: done

File: app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt (hunk 5 above)
Comment (bhm) [reply app/ui/src/main/java/com/fsck/k9/fragment/MessageListAdapter.kt]: done

**Comment by bhm** [comment]
Yep, just noticed while doing a self review. I've pushed another commit that decouples adapter from K9 dependency further.
I'll give it another look if anything else makes sense to offload to the appearance data class.

**Comment by bhm** [comment]
Colors and backgrounds extraction also got moved to the appearance class. Hopefully this makes sense to you too.

**Comment by cketti** [comment]
> Colors and backgrounds extraction also got moved to the appearance class. Hopefully this makes sense to you too.

I think that's overdoing it. That way you need to pass dependencies to `MessageListAppearance` and have logic in there. That makes it no longer a simple data class.

`MessageListAdapter` is responsible for doing view inflation. I don't see why resolving theme attributes shouldn't also happen there.

**Comment by cketti** [comment]
Thanks
